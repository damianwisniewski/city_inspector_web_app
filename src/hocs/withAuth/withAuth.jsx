import React, { Component, Fragment } from 'react'
import { Redirect } from 'react-router'
import { connect } from 'react-redux'

import { PopupRedirect } from '../../components/PopupComponents/PopupRouting'

export default function withAuth(WrappedComponent) {
	const mapStateToProps = state => ({
		isUserAuth: state.user.isUserAuth,
	})

	return connect(mapStateToProps)(
		class extends Component {
			render() {
				return this.props.isUserAuth ? (
					<WrappedComponent {...this.props} />
				) : (
					<Fragment>
						<Redirect to='/' />
						<PopupRedirect
							message='Nie masz dostępu do tych zasobów. Aby korzystać w pełni z aplikacji musisz się zalogować!'
							to='LoginPopup'
						/>
					</Fragment>
				)
			}
		},
	)
}
