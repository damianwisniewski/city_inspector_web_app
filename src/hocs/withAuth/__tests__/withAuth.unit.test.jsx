import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import withAuth from '../withAuth'

configure({ adapter: new Adapter() })

describe('withAuth() HOC', () => {
	const FakeComponent = withAuth(() => <div>TEST</div>)

	test('should render Component for authorized user', () => {
		const wrapper = shallow(<FakeComponent.WrappedComponent />)
		wrapper.setProps({ isUserAuth: true })
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	test('should render Redirections for non-authorized user', () => {
		const wrapper = shallow(<FakeComponent.WrappedComponent />)
		wrapper.setProps({ isUserAuth: false })
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})
