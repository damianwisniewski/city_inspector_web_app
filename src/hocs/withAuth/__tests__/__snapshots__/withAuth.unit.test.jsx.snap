// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withAuth() HOC should render Component for authorized user 1`] = `
<Component
  isUserAuth={true}
/>
`;

exports[`withAuth() HOC should render Redirections for non-authorized user 1`] = `
<Fragment>
  <Redirect
    to="/"
  />
  <PopupRedirect
    message="Nie masz dostępu do tych zasobów. Aby korzystać w pełni z aplikacji musisz się zalogować!"
    to="LoginPopup"
  />
</Fragment>
`;
